gridftp/net_manager/python/globus_net_manager_python.c

```diff
@@ -353,40 +358,51 @@ globus_l_net_manager_python_handle_exception(
     const char                         *func_name)
 {
     globus_result_t                     result = GLOBUS_SUCCESS;
-    PyObject                           *exception = NULL,
-                                       *exception_string = NULL;
-    const char                         *exception_cstr = NULL;
-
-    exception = PyErr_Occurred();
-    if (exception != NULL)
+    PyObject                           *exc_type = NULL,
+                                       *exc_value = NULL,
+                                       *exc_trace = NULL,
+                                       *exc_type_string = NULL,
+                                       *exc_value_string = NULL;
+    const char                         *exc_type_cstr = NULL,
+                                       *exc_value_cstr = NULL;
+
+    if (PyErr_Occurred())
     {
-        exception_string = PyObject_Str(exception);
-        if (exception_string)
+        PyErr_Fetch(&exc_type, &exc_value, &exc_trace);
+        PyErr_NormalizeException(&exc_type, &exc_value, &exc_trace);
+        exc_type_string = PyObject_Str(exc_type);
+        exc_value_string = PyObject_Str(exc_value);
+        PyErr_Restore(exc_type, exc_value, exc_trace);
+
+        if (exc_type_string && exc_value_string)
         {
-            exception_cstr = PyString_AsString(exception_string);
-            if (exception_cstr)
+            exc_type_cstr = PyString_AsString(exc_type_string);
+            exc_value_cstr = PyString_AsString(exc_value_string);
+            if (exc_type_cstr && exc_value_cstr)
             {
                 char *error_explanation = globus_common_create_string(
-                        "Python exception in %s: %s",
+                        "%s exception in %s: %s",
+                        exc_type_cstr,
                         func_name,
-                        exception_cstr);
+                        exc_value_cstr);
 
                 if (error_explanation != NULL)
                 {
                     result = GlobusNetManagerErrorManager(
-                            GLOBUS_FAILURE,
+                            GLOBUS_SUCCESS,
```

Comment:
Why GLOBUS_SUCCESS?

Reply:
Might be good to add a short comment in the code, either here or elsewhere. It is 'counter-intuitive' to say the least...

Reply:
Would it be better to write:
result = GlobusNetManagerErrorManager(result, ....)
instead of:
result = GlobusNetManagerErrorManager(GLOBUS_SUCCESS, ....) ?
I think this would make it more clear that this extends the existing result value without hard-coding it.

Reply:
Yes, I think that's clear enough, thanks!